File: Exam3RegladeNegocio/Js/ec/edu/espe/payroll/utils/MongoConnection.js
const { MongoClient } = require('mongodb');

class MongoConnection {
    constructor() {
        if (!MongoConnection.instance) {
            // TU CADENA DE CONEXIÓN EXACTA
            this.uri = "mongodb+srv://Mikael:Jaime2006@cluster0.fpyoe9m.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
            this.client = new MongoClient(this.uri);
            this.dbName = "payrollHidalgo";
            this.connectionPromise = this.client.connect(); // Conectamos una sola vez
            MongoConnection.instance = this;
            console.log("--> Instancia Singleton de MongoDB Creada <--");
        }
        return MongoConnection.instance;
    }

    async getCollection(collectionName) {
        await this.connectionPromise; // Esperar a que conecte
        return this.client.db(this.dbName).collection(collectionName);
    }
}

// Inicializamos la instancia
const instance = new MongoConnection();
Object.freeze(instance); // Congelamos para que nadie la modifique

module.exports = instance;

File: Exam3RegladeNegocio/Python/ec/edu/espe/payroll/utils/mongo_connection.py
from pymongo import MongoClient

class MongoConnection:
    _instance = None
    _client = None
    _db = None

    # El patrón Singleton en Python: __new__ garantiza una sola instancia
    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(MongoConnection, cls).__new__(cls)
            try:
                # TU CADENA DE CONEXIÓN EXACTA
                uri = "mongodb+srv://Mikael:Jaime2006@cluster0.fpyoe9m.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"
                cls._client = MongoClient(uri)
                cls._db = cls._client["payrollHidalgo"]
                print("Conexión Singleton establecida.")
            except Exception as e:
                print(f"Error de conexión: {e}")
        return cls._instance

    def get_collection(self, collection_name):
        return self._db[collection_name]